rs/ic_os/config/src/main.rs

```diff
@@ -0,0 +1,157 @@
+use anyhow::Result;
+use clap::{Parser, Subcommand};
+use config::config_ini::get_config_ini_settings;
+use config::deployment_json::get_deployment_settings;
+use config::serialize_and_write_config;
+use std::fs::File;
+use std::path::{Path, PathBuf};
+
+use config::types::{
+    GuestOSSettings, HostOSConfig, HostOSSettings, ICOSSettings, Logging, NetworkSettings,
+    SetupOSConfig, SetupOSSettings,
+};
+
+#[derive(Subcommand)]
+pub enum Commands {
+    /// Creates SetupOSConfig object
+    CreateSetuposConfig {
+        #[arg(long, default_value = config::DEFAULT_SETUPOS_CONFIG_FILE_PATH, value_name = "config.ini")]
+        config_ini_path: PathBuf,
+
+        #[arg(long, default_value = config::DEFAULT_SETUPOS_DEPLOYMENT_JSON_PATH, value_name = "deployment.json")]
+        deployment_json_path: PathBuf,
+
+        #[arg(long, default_value = config::DEFAULT_SETUPOS_NNS_PUBLIC_KEY_PATH, value_name = "nns_public_key.pem")]
+        nns_public_key_path: PathBuf,
+
+        #[arg(long, default_value = config::DEFAULT_SETUPOS_SSH_AUTHORIZED_KEYS_PATH, value_name = "ssh_authorized_keys")]
+        ssh_authorized_keys_path: PathBuf,
+
+        #[arg(long, default_value = config::DEFAULT_SETUPOS_NODE_OPERATOR_PRIVATE_KEY_PATH, value_name = "node_operator_private_key.pem")]
+        node_operator_private_key_path: PathBuf,
+
+        #[arg(long, default_value = config::DEFAULT_SETUPOS_CONFIG_OBJECT_PATH, value_name = "config.json")]
+        setupos_config_json_path: PathBuf,
+    },
+    /// Creates HostOSConfig object from existing SetupOS config.json file
+    GenerateHostosConfig {
+        #[arg(long, default_value = config::DEFAULT_SETUPOS_CONFIG_OBJECT_PATH, value_name = "config.json")]
+        setupos_config_json_path: PathBuf,
+    },
+}
+
+#[derive(Parser)]
+#[command()]
+struct ConfigArgs {
+    #[command(subcommand)]
+    command: Option<Commands>,
+}
+
+pub fn main() -> Result<()> {
+    let opts = ConfigArgs::parse();
+
+    match opts.command {
+        Some(Commands::CreateSetuposConfig {
+            config_ini_path,
+            deployment_json_path,
+            nns_public_key_path,
+            ssh_authorized_keys_path,
+            node_operator_private_key_path,
+            setupos_config_json_path,
+        }) => {
+            // get config.ini settings
+            let config_ini_settings = get_config_ini_settings(&config_ini_path)?;
+
+            // get deployment.json variables
+            let deployment_json_settings = get_deployment_settings(&deployment_json_path)?;
+
+            let network_settings = NetworkSettings {
```

Comment:
Note: On type boundaries like this, unpacking the structs entirely first can help catch unused fields:
```
let ConfigIniSettings {
    ipv6_prefix,
    ipv6_address,
    ipv6_prefix_length,
    ipv6_gateway,
    ipv4_address,
    ipv4_gateway,
    ipv4_prefix_length,
    domain,
    verbose,
} = config_ini_settings;
```